pkg/workspace/apply_test.go

```diff
@@ -522,3 +524,310 @@ func TestApply(t *testing.T) {
 		})
 	}
 }
+
+func TestApply_IsolatedWorkspaces(t *testing.T) {
```

Comment:
NIT: The tests included here are pretty comprehensive (go coverage is very obvious) 😍  but will be great to have a test with two different workspaces out of which one is isolated to a step and one is isolated to a sidecar.